pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongHashMap.java

```diff
@@ -361,6 +361,9 @@ V get(long key, int keyHash) {
         }
 
         V put(long key, V value, int keyHash, boolean onlyIfAbsent, LongFunction<V> valueProvider) {
+            if (value == null) {
+                requireNonNull(valueProvider.apply(key));
```

Comment:
Even if modification is required, we should only run `valueProvider.apply(key)` once. reason :
1. The two calculated values may be different
2. cost performance

Reply:
how about this ?
![image](https://user-images.githubusercontent.com/39521534/208330122-e43a8e37-1d48-4bb8-b4bd-7518e3fb71ab.png)

Reply:
> 

yes, we should run valueProvider.apply(key) once.  like above,

Reply:
I'll discuss it here（https://github.com/apache/bookkeeper/pull/3706） and sync it up after the merge, thank you for your review, @congbobo184